File: app/src/main/java/com/textfield/json/outabus/Trip.java
package com.textfield.json.outabus;

import java.util.ArrayList;

/**
 * Created by Jason on 19/04/2016.
 */
public class Trip {
    String id, destination;
    ArrayList<Stop> stops;

    public Trip(Bus bus, ArrayList<Stop> stops){
    }
    public Trip(String id){
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        return (stops.get(0).equals(((Trip)o).stops.get(0)) && stops.get(stops.size()-1).equals(((Trip)o).stops.get(stops.size()-1)));
    }

}

File: app/src/main/java/com/textfield/json/outabus/entities/Trip.kt
package com.textfield.json.outabus.entities

import java.util.ArrayList

/**
 * Created by Jason on 19/04/2016.
 */
class Trip {
    internal var id = ""
    internal var destination = ""
    internal var stops = ArrayList<Stop>()

    constructor(bus: Bus, stops: ArrayList<Stop>) {}
    constructor(id: String) {
        this.id = id
    }

    override fun equals(other: Any?): Boolean {
        return stops[0] == (other as Trip).stops[0] && stops[stops.size - 1] == other.stops[stops.size - 1]
    }

}
